src/main/java/uk/gov/hmcts/ccd/service/MessagePublisherRunnable.java

```diff
@@ -59,14 +62,35 @@ private void processUnpublishedMessages(Pageable pageable,
         } else if (!processedEntities.isEmpty()) {
             messageQueueCandidateRepository.saveAll(processedEntities);
             log.info(String.format("%s Published %s messages to destination '%s'",
-                logPrefix, processedEntities.size(), publishMessageTask.getDestination()));
+                                   logPrefix, processedEntities.size(), publishMessageTask.getDestination()
+            ));
         }
     }
 
+    private String getPropertyValue(JsonNode data, String propertyId) {
+
+        return data.get(propertyId).asText();
+    }
+
+    private Message setProperties(Message message, JsonNode data) throws JMSException {
+        for (MessageProperties property : MessageProperties.values()) {
+            if ((data.has(property.getPropertyId())) && (!(getPropertyValue(
+                data, property.getPropertyId()).equals("null")))) {
+                message.setStringProperty(property.getPropertySourceId(), getPropertyValue(
+                    data, property.getPropertyId())
```

Comment:
Not sure if `propertyId`/`propertySourceId` are the wrong way round logically?
```suggestion
            if (data.has(property.getPropertySourceId() && !getPropertyValue(
                data, property.getPropertySourceId()).equals("null")) {
                message.setStringProperty(property.getPropertyId(), getPropertyValue(
                    data, property.getPropertySourceId())
```